tests/product/test_server_install.py

```diff
@@ -191,12 +191,12 @@ def test_install_with_java8_home(self, dummy=True):
             self.assert_has_default_config(container)
             self.assert_has_default_connector(container)
 
-    def test_install_failure_without_java8_home(self, dummy=True):
+    def test_install_failure_without_java8_home(self):
         for container in self.cluster.all_hosts():
             self.cluster.exec_cmd_on_host(container,
                                           "mv /usr/java/jdk1.8.0_40 /usr/")
         self.upload_topology()
-        cmd_output = self.installer.install(dummy)
+        cmd_output = self.installer.install(dummy=True, pa_raise_error=False)
```

Comment:
The way server install used to work masked the error because it re-executed the task and (presumably) the return value of that wasn't propagated all the way back to main.py where exit_code would notice.

Since server install no longer execute()s deploy_install, errors show up as non-zero exit codes, which raise OSErrors if you don't add this.